### PR TITLE
fix apiserver pprof redirect bug

### LINKED ISSUE #66812: apiserver pprof redirect bug
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature

**What happened**:
The links on  **http://${APISERVER}:8080/debug/pprof** will not jump to a correct address.
For example, if you click [goroutine](http://127.0.0.1:8080/debug/goroutine?debug=1), it will jump to  **http://${APISERVER}:8080/debug/goroutine?debug=1**. 

**What you expected to happen**:
The above example will redirect to **http://${APISERVER}::8080/debug/pprof/goroutine?debug=1** which will show all the info of goroutines.

**How to reproduce it (as minimally and precisely as possible)**:
Find an apiserver and try it out.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
**v1.9.6**
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
**go version go1.10.3**
- Others:

### PR DESCRIPTION
**Pull request by DylanBLE**
**What this PR does / why we need it**:
fix apiserver pprof redirect bug

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #66812

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by DylanBLE** [comment]
@wgliang  fixed

File: staging/src/k8s.io/apiserver/pkg/server/mux/pathrecorder.go [hunk 1]
```diff
@@ -276,3 +276,10 @@ func (s byPrefixPriority) Less(i, j int) bool {
 
 	return strings.Compare(s[i], s[j]) < 0
 }
+
+// RedirectTo redirects request to a certain destination.
+func RedirectTo(to string) func(http.ResponseWriter, *http.Request) {
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/mux/pathrecorder.go]: this does not look like something that belongs into `mux`. Better turn it into a local helper in staging/src/k8s.io/apiserver/pkg/server/routes

Comment (DylanBLE) [reply staging/src/k8s.io/apiserver/pkg/server/mux/pathrecorder.go]: fixed.

File: staging/src/k8s.io/apiserver/pkg/server/routes/helper.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+*/
+
+package routes
+
+import (
+	"net/http"
+)
+
+// RedirectTo redirects request to a certain destination.
+func RedirectTo(to string) func(http.ResponseWriter, *http.Request) {
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/routes/helper.go]: you can leave it in the old file, but make it lower case.

Comment (DylanBLE) [reply staging/src/k8s.io/apiserver/pkg/server/routes/helper.go]: I changed RedirectTo to lower case.

File: staging/src/k8s.io/apiserver/pkg/server/routes/helper.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+*/
+
+package routes
+
+import (
+	"net/http"
+)
+
+// redirectTo redirects request to a certain destination.
+func redirectTo(to string) func(http.ResponseWriter, *http.Request) {
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/routes/helper.go]: and please move it back into profiling.go. Having one-time-used helpers of 5 lines in a separate 28 line file looks like overkill.

Comment (DylanBLE) [reply staging/src/k8s.io/apiserver/pkg/server/routes/helper.go]: got it, thanks for your comments~

**Pull request merged**